app/latigo/types.py

```diff
@@ -56,6 +56,16 @@ class SensorDataSet:
     data: typing.List[pd.Series]
     meta_data: typing.Dict = field(default_factory=dict)
 
+    def __eq__(self, other):
+        if self.time_range != other.time_range:
+            return False
+        if self.meta_data != other.meta_data:
+            return False
+        for i, series in enumerate(self.data):
+            if not series.equals(other.data[i]):
+                return False
```

Comment:
There should be a check for a case when `self.data` and `other.data` has a different number of elements.

Also current approach could be rewritten with `zip`. Putting everything together:

```suggestion
        if len(self.data) != len(other.data):
            return False

        if not all(s1.equals(s2) for s1, s2 in zip(self.data, other.data)):
            return False
```

Or:

```suggestion
        if len(self.data) != len(other.data):
            return False

        for self_series, other_series in zip(self.data, other.data)):
            if not series.equals(other_series):
                return False
```

Reply:
Nice, this way is cleaner :+1: